Test failures appear to be unrelated to my changes. Please let me know if that is incorrect.

Example failure:
```
======================================================================
ERROR: IPython.core.tests.test_inputtransformer2.test_check_complete
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.8/x64/lib/python3.9/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/runner/work/ipython/ipython/IPython/core/tests/test_inputtransformer2.py", line 270, in test_check_complete
    assert cc("a \\ ") == ("invalid", None)  # Nothing allowed after backslash
  File "/home/runner/work/ipython/ipython/IPython/core/inputtransformer2.py", line 753, in check_complete
    res = compile_command(''.join(lines), symbol='exec')
  File "/opt/hostedtoolcache/Python/3.9.8/x64/lib/python3.9/codeop.py", line 178, in __call__
    return _maybe_compile(self.compiler, source, filename, symbol)
  File "/opt/hostedtoolcache/Python/3.9.8/x64/lib/python3.9/codeop.py", line 98, in _maybe_compile
    code2 = compiler(source + "\n\n", filename, symbol)
  File "/home/runner/work/ipython/ipython/IPython/core/inputtransformer2.py", line 786, in __call__
    return compile(*args, **kwds)
SystemError: Negative size passed to PyUnicode_New
```

--- a/IPython/terminal/interactiveshell.py
+++ b/IPython/terminal/interactiveshell.py
@@ -505,7 +505,7 @@ class TerminalInteractiveShell(InteractiveShell):
         # while/true inside which will freeze the prompt.
 
         try:
-            old_loop = asyncio.get_event_loop()
+            old_loop = asyncio.get_running_loop()
         except RuntimeError:
             # This happens when the user used `asyncio.run()`.
             old_loop = None

